### DISCUSSION TITLE
Balancer: how to select other outbounds only when the 1st has failed?

### DISCUSSION THREAD

**Discussion by pulsarice** [XTLS/Xray-core#3068] [Q&A]
Hi
I'm looking for a solution to have high availability, But I don't want to select outbounds randomly or based on leastLoad or leastPing.
I want xray to use the first outbound as long as it pings (even if it is lower than the others) and only if the first one has failed, choose the others in selector list.
The outbounds are Portal, so I can't use vnext.

**Comment by pulsarice** [comment]
I have found the answer. I need to use a loopback outbound. 
balancer1 : only contains one selector with fallbackTag set to loopback outbound
loopback inbound goes to balancer2 via routing rules.
balancer2: contains all other selectors.
balancers should have either leastPing strategy with observatory configuration or leastLoad strategy with burstObservatory configuration.

**Reply by firengate** [reply]
So it turns out that if you have, say, 10 proxies, you need 10 balancers, 10 route rules and 10 loopbacks?
